### PR TITLE
161780577 Bookmarking Articles

### PR DESCRIPTION
**Pull request by Fiyiin**
#### Description
This PR adds the functionality for users to add or remove bookmarks from articles they intend to read later

#### Type of Change
- Feature
  - creates an endpoint to add bookmarks
  - creates an endpoint to get bookmarks for a user
  - creates an endpoint to remove bookmarks
  - add tests

#### Pivotal Tracker
 #161780577

#### Flowchart
Flowchart for Bookmarks-2

### CODE REVIEW

**Comment by kossy360** [comment]
LGTM

**Comment by Fiyiin** [comment]
Flowchart
Flowchart for Bookmarks

File: src/routes/api/bookmarks/bookmarkController.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      await bookmark.destroy({
+        where: { user, article }
+      });
+      return res.status(200).json({
+        status: 200,
+        message: 'Bookmark removed'
+      });
+    } catch (err) {
+      return res.status(500).json({
+        status: 500,
```
Comment (profmcdan) [inline src/routes/api/bookmarks/bookmarkController.js]: The error handling here is ambiguous. KIndly review this code and provide a more robust and descriptive error handling functionality.

**Review by profmcdan — CHANGES_REQUESTED** [review]
Please implement the required changes

**Comment by profmcdan** [comment]
Review the flowchart design

**Comment by Fiyiin** [comment]
Flowchart for Bookmarks-2

File: src/routes/api/bookmarks/bookmarkController.js (hunk 1 above)
Comment (Fiyiin) [reply src/routes/api/bookmarks/bookmarkController.js]: Fixed.

**Review by profmcdan — APPROVED** [review]

**Pull request merged**